db/views.c

```diff
@@ -792,12 +792,16 @@ static int _generate_new_shard_name(const char *oldname, char *newname,
         /* get the length of the number sufix */
         suffix_len = _shard_suffix_str_len(maxshards);
 
-        snprintf(newname, newnamelen, "%s%.*d", oldname, suffix_len, nextnum);
+        if (0 > snprintf(newname, newnamelen, "%s%.*d", oldname, suffix_len, nextnum)) {
```

Comment:
Thanks for the pull request! I think it should check for `snprintf(...) >= newnamelen`, instead. snprintf returns number of bytes written. If output is truncated, it returns number of bytes which would've been written as if output had not been truncated. So any return value >= `newnamelen` means truncation.